test/functional/p2p_invalid_block.py

```diff
@@ -133,5 +139,19 @@ def run_test(self):
         self.log.info("Test inflation by duplicating input")
         peer.send_blocks_and_test([block4], node, success=False,  reject_reason='bad-txns-inputs-duplicate')
 
+        self.log.info("Test accepting identical block after rejecting it due to a future timestamp.")
+        t = int(time.time())
+        # Make block timestamp validity 60 seconds premature to avoid test-timing issues
```

Comment:
why? I changed to `1` and nothing happened

Reply:
I also had no difficulty/local failures with t=1, but there was a raised concern that multi-threaded tests could jeopardise this [here](https://github.com/bitcoin/bitcoin/pull/22120#discussion_r643774433). It doesn't seem like something that threat timing issues could throw off to me as time is advanced before validity is re-checked, but I am not 100% on that.

I am going to reset this to 1 along with the timeout change suggested as a means to restart appveyor which failed.

Reply:
> Edit: @MarcoFalke can you confirm that node time cannot advance automatically once setmocktime rpc has been called? If so then I will be convinced that multi-threading these tests will not cause any issues with t=1

The wall clock time for the node still advances, but the mocktime is pinned once `setmocktime` is set to a non-zero value.

Reply:
Thanks, unnecessary constant now removed in 754e802